src/test/headers_sync_chainwork_tests.cpp

```diff
@@ -234,4 +244,28 @@ static void TooLittleWork(const CBlockIndex* chain_start,
         /*exp_locator_hash=*/std::nullopt);
 }
 
+static void TooBigBuffer(const CBlockIndex* chain_start,
+        const std::vector<CBlockHeader>& first_chain)
+{
+    // Intentionally too big redownload buffer in order to trigger warning.
+    HeadersSyncState hss{CreateState(chain_start,
+                                     HeadersSyncParams{
+                                         .commitment_period = COMMITMENT_PERIOD,
+                                         .redownload_buffer_size = first_chain.size(),
+                                     })};
+
+    CHECK_RESULT(hss.ProcessNextHeaders(first_chain, true),
+        hss, /*exp_state=*/State::REDOWNLOAD,
+        /*exp_success*/true, /*exp_request_more=*/true,
+        /*exp_headers_size=*/0, /*exp_pow_validated_prev=*/std::nullopt,
+        /*exp_locator_hash=*/Params().GenesisBlock().GetHash());
```

Comment:
might make sense to extract genesis hash here as well like we did in `HappyPath`